src/SourceBuild/tarball/patches/runtime/0002-Eliminate-usages-of-pre-built-packages-during-source.patch

```diff
@@ -0,0 +1,188 @@
+From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
+From: Eric Erhardt <eric.erhardt@microsoft.com>
+Date: Tue, 12 Oct 2021 19:40:56 +0000
+Subject: [PATCH 2/2] Eliminate usages of pre-built packages during
+ source-build
+
+* Don't build MSBuild tasks for netfx during source-build
+* Don't build Microsoft.Windows.Compatibility during source-build
+* Don't build test projects during source-build
+* Don't restore Microsoft.DiaSymReader.Native
+* Don't build source generators that target older Roslyn versions
```

Comment:
> are these source generators that ship in the framework?

See https://github.com/dotnet/sdk/issues/20355 for more information on what we are doing here.

In short, we are building our source-generators twice, each targeting different Roslyn versions:

* 3.11
* 4.0

The 4.0 targeted version ships in the framework (we are still building the 4.0 versions in source-build). And they both ship in the NuGet packages.

The reason we want to do this is so we can use new APIs introduced in 4.0, but keep our source-generators working in VS 2019 which uses the 3.x Roslyn version. The 3.x versions are slower and don't support new language features (e.g. file-scoped namespaces).

> Would not building these prevent someone who is using source-build with a manually-lowered Roslyn version (acquired from NuGet.org) from building a repo that relies on the source generators? Or is that scenario impossible?

That scenario is impossible to build dotnet/runtime because we are using new language features that only came in Roslyn 4.0. It also would mean that the source generators that ship in the framework couldn't be built, since they need the new Roslyn APIs. Lastly, in order to target `net6.0`, you need to use the new Roslyn version.